packages/issuer/src/blockchain-facade/Certificate.ts

```diff
@@ -0,0 +1,257 @@
+    public deviceId: string;
+
+    public generationStartTime: number;
+    public generationEndTime: number;
+    public issuer: string;
+    public energy: number;
+    public creationTime: number;
+
+    public initialized: boolean;
+
+    private data: number[];
+
+    constructor(id: string, configuration: Configuration.Entity) {
+        super(id, configuration);
+
+        this.initialized = false;
+    }
+
+    async sync(): Promise<Entity> {
+        if (this.id === null) {
+            return this;
+        }
+
+        const registry: Registry = this.configuration.blockchainProperties.certificateLogicInstance;
+        const publicIssuer: PublicIssuer = this.configuration.blockchainProperties.issuerLogicInstance.public;
+
+        const cert = await registry.getCertificate(Number(this.id));
+
+        this.data = cert.data;
+
+        const decodedData = await publicIssuer.decodeIssue(this.data);
+
+        this.generationStartTime = Number(decodedData['0']);
+        this.generationEndTime = Number(decodedData['1']);
+        this.deviceId = decodedData['2'];
+
+        this.issuer = cert.issuer;
+
+        const allIssuanceEvents = await registry.getAllIssuanceSingleEvents({
+            filter: { _id: Number(this.id) }
+        });
+        const creationBlock = await this.configuration.blockchainProperties.web3.eth.getBlock(allIssuanceEvents[0].blockNumber);
+
+        this.creationTime = Number(creationBlock.timestamp);
+
+        this.initialized = true;
+
+        if (this.configuration.logger) {
+            this.configuration.logger.verbose(`Certificate ${this.id} synced`);
+        }
+
+        return this;
+    }
+
+    async getCertificateOwner(): Promise<string> {
+        return this.configuration.blockchainProperties.certificateLogicInstance.getCertificateOwner(
+            this.id
+        );
+    }
+
+    async claim(amount: number): Promise<TransactionReceipt> {
+        const registry: Registry = this.configuration.blockchainProperties.certificateLogicInstance;
+        const owner = this.configuration.blockchainProperties.activeUser.address;
+
+        const { randomHex, hexToBytes } = this.configuration.blockchainProperties.web3.utils;
+
+        const claimData = randomHex(32);
+
+        return registry.safeTransferAndClaimFrom(
+            owner,
+            owner,
+            parseInt(this.id, 10),
+            amount,
+            this.data,
+            hexToBytes(claimData),
+            getAccountFromConfiguration(this.configuration)
+        );
+    }
+
+    async transfer(to: string, amount: number): Promise<TransactionReceipt> {
+        const registry: Registry = this.configuration.blockchainProperties.certificateLogicInstance;
+        const from = this.configuration.blockchainProperties.activeUser.address;
+        
+        return registry.safeTransferFrom(
+            from,
+            to,
+            parseInt(this.id, 10),
+            Math.round(amount),
```

Comment:
What is the reason for rounding here?

Reply:
There's no way to transfer a part of a token in "ERC20". This rounds it to avoid a `revert`, but you're right, this might not be desired behaviour.